### ISSUE TITLE
Message threads only showing most recent message until first usage

### ISSUE THREAD

**Issue by ThePatrickHe** [GSConnect/gnome-shell-extension-gsconnect#1625]
### Describe the bug

My problem is identical to this Reddit post: https://www.reddit.com/r/gnome/comments/11aglp4/gsconnect_issues_when_replying_to_messages/

When initially connecting my phone to my PC, all of my message threads and contacts load in. However, only the most recent message in each thread is displayed. When I open a thread and send a message using GSConnect, that thread suddenly populates with full message history, but all the other threads disappear for about a minute. Then, the other threads return (still showing only the most recent message).

### Steps to reproduce

1. Download GSConnect rpm application from Gnome Software.
2. Download GSConnect extension using Extension Manager application from Gnome Software.
3. Download KDE Connect application from F-droid on phone.
4. Pair devices as normal
5. Open messaging on GSConnect.
6. Send a message.

### Expected behavior

Upon first connection, there should be a small delay as messages sync, then all messages should automatically sync. 

### GSConnect version

55

### Installed from

Extension Manager

### GNOME Shell version

44.1

### Linux distribution/release

Fedora 38

### Paired device(s)

Samsung S20 5G

### KDE Connect app version

1.24.5

### Plugin(s)

All except for `Receive notifications` and `Receive remote keypresses` and `Notifications sync`

**Comment by ferdnyc** [comment]
@ThePatrickHe 

Yeah, it seems there [was a change made](https://invent.kde.org/network/kdeconnect-android/-/merge_requests/296) to how KDEConnect returns message results, that messes with our cache-population logic.

Previously, the results for `kdeconnect.sms.request_conversations` came back with all of the threads' most recent messages together in one message packet. When we detect a response containing multiple threads, it's processed as a `request_threads` dump, so we flush the cache of message threads and use the packet's contents to set up a new one. Then we walk all of the threads one by one, requesting each thread's history to fill the cache with messages.

Since that change I linked above, the results for the `request_conversations` packet are sent back one-at-a-time: each thread's latest-message data comes in a separate `kdeconnect.sms.messages` packet all on its own, and we end up getting N separate packets for the N available threads.

That's fine, in terms of receiving the data, but it breaks our cache-filling logic. Because only a KDEConnect message packet that contains _more than one thread_ will trigger the cache update — and KDEConnect never sends packets like that anymore, in response to `request_conversations` — it'll never signal GSConnect to start filling its cache.

But it gets even worse. Here's where things get **really** stupid:

When you _send_ a text via the Messaging interface, that action always triggers a new-message update from KDEConnect. (Because a new text message _has_ arrived: The one you just sent. Outgoing messages are treated the same as incoming messages, from the SMS database perspective.)

There's a quirk in the way that's handled by KDEConnect, though. Whenever it sends messages to us, KDEConnect keeps track of the `mostRecentTimestamp` of the included messages. That way, whenever it has to send us **_new_** messages, it can just grab everything in the database with a timestamp equal to or greater than its cached `mostRecentTimestamp`, dump it all into a packet, and send it our way.

Because that timestamp check is **_EQUAL TO_ or greater than** the `mostRecentTimestamp`, the previously-sent most recent message (the one that set the `mostRecentTimestamp`) will be included in the response. So, there will always be one extra message included with the new messages.

Which means that, after you send a text, **two** messages come back in the new-message packet: The previously most-recent text, plus the new text you just sent. And unless the thread you sent your message to also happened to be the most-recent-timestamp thread, that new-message packet will contain two messages from two different threads.

Remember that a packet containing more than one thread is the exact thing (the _only_ thing) that triggers a cache update in GSConnect.

So when that multi-thread (two-thread) packet comes back in response to your sent text, GSConnect throws away the entire message cache _except_ for the threads contained in that packet!

Obviously, our cache logic has to be completely gutted and redesigned, in light of this KDEConnect change.

**Comment by kohend** [comment]
Is this something that an external contributor like me can contribute? Or does this require a decision in the project level on how to do it and until a decision is reached, you won't accept a PR for this logic change?

**Comment by andyholmes** [comment]
> does this require a decision in the project level

Nope, go right ahead! The solution can't be *bonkers* of course, but otherwise have at it.

**Comment by ferdnyc** [comment]
I should note that I've made some _inroads_ on an attempted implementation of cache-less SMS support. It:

1. Does away with the JSON cache entirely
2. Caches only the `request_threads` results, in memory. (And I'm thinking of doing away with that, too, in favor of populating it only when Messages is actually opened.)
3. Populates each thread in the Messages interface only on-demand, if/when when it's actually viewed.
4. Loads in thread messages 20 at a time, in reverse order. (I'd also like to get that working where it only further back-populates a thread if the user scrolls to the top of the current history.)
5. Effectively uses the Messages window itself as the cache, since whatever messages have been loaded into each thread there will remain populated until the window is destroyed.

I've never been happy with GSConnect's aggressive SMS caching. Even when it worked. The basic problem is, it causes an _endless_ parade of downloads for messages that **I DON'T EVEN WANT** locally.

The SMS database on my phone is a wasteland of automated messages, ancient conversations from years ago, and spam texts with images attached (which, at least on my Samsung phones, are always included in the message data despite not even being displayable by GSConnect currently). Meaning that every time my phone connects to GSConnect, it starts re-shoveling the same tens of megabytes of crap over the link _again_. When 99% of the time, I'll never even open the Messages window at all.

The code definitely isn't working quite right yet, but it's sort of getting there. I noticed that I hadn't even pushed a copy up to my fork here, so I did that just now. It's at https://github.com/ferdnyc/gnome-shell-extension-gsconnect/tree/no-cache-sms
